### ISSUE TITLE
Changing google_alloydb_cluster database_version has unexpected plan/execution

### ISSUE THREAD

**Issue by wjessop** [hashicorp/terraform-provider-google#17403]
### Community Note

* Please vote on this issue by adding a 👍 [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request.
* Please do not leave _+1_ or _me too_ comments, they generate extra noise for issue followers and do not help prioritize the request.
* If you are interested in working on this issue or have submitted a pull request, please leave a comment.
* If an issue is assigned to a user, that user is claiming responsibility for the issue.
* Customers working with a Google Technical Account Manager or Customer Engineer can ask them to [reach out internally](https://github.com/hashicorp/terraform-provider-google/wiki/Customer-Contact#raising-gcp-internal-issues-with-the-provider-development-team)
  to expedite investigation and resolution of this issue.

### Terraform Version

Terraform v1.3.5
on darwin_arm64
+ provider registry.terraform.io/cyrilgdn/postgresql v1.21.0
+ provider registry.terraform.io/hashicorp/archive v2.4.2
+ provider registry.terraform.io/hashicorp/aws v5.38.0
+ provider registry.terraform.io/hashicorp/google v5.17.0
+ provider registry.terraform.io/hashicorp/random v3.6.0

### Affected Resource(s)

google_alloydb_cluster

### Terraform Configuration

```tf
resource "google_alloydb_cluster" "full" {
  cluster_id     = "widgets"
  location       = "us-east4"

  # New config added to existing database
  database_version = "POSTGRES_15"

  # …
}
```

### Expected Behavior

When setting the version of an AlloyDB postgres database to be an new version compared to the existing running database Terraform should prompt to destroy and re-create the database as an in-place upgrade is not possible.

### Actual Behavior

Terraform reports an apply operation completed successfully but the database remains the old version in the web UI.

### Steps to reproduce

1. `terraform apply`

### References

database_version was introduced here https://github.com/hashicorp/terraform-provider-google/issues/16960

**Comment by zli82016** [comment]
Based on the documentation, `database_version` cannot be changed after the cluster creation.  

https://registry.terraform.io/providers/hashicorp/google/latest/docs/resources/alloydb_cluster

**Comment by wjessop** [comment]
@zli82016 No, I knew that. It doesn't make sense that Terraform would just complete with no mention of the fact that something was configured to change that ultimately can't be and won't be changed.

The change requires the cluster be re-built so it is expected that terraform should prompt to re-build the cluster with a "forces replacement".

**Comment by zli82016** [comment]
The  issue will be forwarded to the alloydb service team.
